src/main/java/org/commonjava/util/partyline/impl/infinispan/model/InfinispanJFS.java

```diff
@@ -150,29 +180,37 @@ FileMeta getMetadata( final File target, final LocalLockOwner owner ) throws IOE
         {
             return lockManager.reentrantSynchronous( path, ( opLock ) -> {
 
-                // we need to use cache.getAdvancedCache().getTransactionManager() to lock this entry and then update the node-level locks.
-                // this will be where we reserve the lock level (or change it, if our node is already represented)
-                FileMeta meta = metadataCache.computeIfAbsent( path, ( p ) -> new FileMeta( p, target.isDirectory(),
-                                                                                            owner ) );
+                TransactionManager transactionManager = metadataCache.getAdvancedCache().getTransactionManager();
+                transactionManager.begin();
+                FileMeta meta = metadataCache.computeIfAbsent( path, ( p ) -> new FileMeta( p, target.isDirectory() ) );
+                // Check the current lock level
+                // TODO: Negotiate the lock level
+                LockLevel currentLockLevel = meta.getLockLevel( this.nodeKey );
```

Comment:
I think we can assert / write the lock level here based on what the LocalLockOwner has. I guess we can do a comparison to see if we need to push the FileMeta back into the cache though.

Three cases:
1. if node hasn't placed a lock, existing == null...add the local lock level and cache.put()
2. if the node has placed a lock and the level == local level, do nothing
3. if the node has placed a lock and the level != local level, replace with local lock level and cache.put()

Also, we probably need a try/catch/finally to handle commit and rollback in a safe way.